base/errorshow.jl

```diff
@@ -610,27 +699,145 @@ function show_reduced_backtrace(io::IO, t::Vector, with_prefix::Bool)
 
+    println(io, "\n--- STACKTRACE ---")
+    stacktrace_linebreaks() && println(io)
+
+    ndigits_max = ndigits(length(t))
+
+    modulecolordict = Dict("" => :default)
+    modulecolorcycler = Iterators.Stateful(Iterators.cycle(STACKTRACE_MODULECOLORS))
+
     push!(repeated_cycle, (0,0,0)) # repeated_cycle is never empty
     frame_counter = 1
     for i in 1:length(displayed_stackframes)
         (frame, n) = displayed_stackframes[i]
-        if with_prefix
-            show_trace_entry(io, frame, n, prefix = string(" [", frame_counter, "] "))
-        else
-            show_trace_entry(io, frame, n)
+
+        print_stackframe(io, frame_counter, frame, ndigits_max, modulecolordict, modulecolorcycler)
+        
+        if i < length(displayed_stackframes)
+            println(io)
+            stacktrace_linebreaks() && println(io)
         end
+
         while repeated_cycle[1][1] == i # never empty because of the initial (0,0,0)
             cycle_length = repeated_cycle[1][2]
             repetitions = repeated_cycle[1][3]
             popfirst!(repeated_cycle)
-            print(io, "\n ... (the last ", cycle_length, " lines are repeated ",
-                  repetitions, " more time", repetitions>1 ? "s)" : ")")
+            printstyled(io,
+                "--- the last ", cycle_length, " lines are repeated ",
+                  repetitions, " more time", repetitions>1 ? "s" : "", " ---", color = :light_black)
+            if i < length(displayed_stackframes)
+                println(io)
+                stacktrace_linebreaks() && println(io)
+            end
             frame_counter += cycle_length * repetitions
         end
         frame_counter += 1
     end
 end
 
+"""
+    print_stackframe(io, i, frame, digit_align_width, modulecolordict::Dict, modulecolorcycler)
+
+Print a stack frame where the module color is determined by looking up the parent module in
+`modulecolordict`. If the module does not have a color, yet, a new one can be drawn
+from `modulecolorcycler`.
+"""
+function print_stackframe(io, i, frame, digit_align_width, modulecolordict, modulecolorcycler)
+    modul = getmodule(frame)
+    parentmodule = split(modul, ".")[1]
+    if !haskey(modulecolordict, parentmodule)
+        modulecolordict[parentmodule] = popfirst!(modulecolorcycler)
+    end
+    modulecolor = modulecolordict[parentmodule]
+
+    print_stackframe(io, i, frame, digit_align_width, modulecolor)
+end
+
+"""
+    print_stackframe(io, i, frame, digit_align_width, modulecolordict::Dict, modulecolorcycler)
+
+Print a stack frame where the module color is set manually with `modulecolor`.
+"""
+function print_stackframe(io, i, frame, digit_align_width, modulecolor)
+
+    file = getfile(frame, stacktrace_expand_basepaths(), stacktrace_contract_userdir())
+    line = getline(frame)
+
+    # add file and line info for accessing frame locations from the repl
+    push!(LAST_SHOWN_LINE_INFOS, (file, line))
```

Comment:
Just mentioning it here: Once #35556 is merged the global `LAST_SHOWN_LINE_INFOS` should be replaced by `get(io, :LAST_SHOWN_LINE_INFOS, Tuple{String, Int}[])`.